### ISSUE TITLE
Cycle 4 funding: uncertainties, masks, quantities, other types of arrays (Marten)

### ISSUE THREAD

**Issue by aaryapatil** [astropy/astropy-project#418]
Issue to track progress of Cycle 4 funding request which was approved in [PR#380](https://github.com/astropy/astropy-project/pull/380).

[Project Description](https://github.com/astropy/astropy-project/blob/main/finance/proposal-calls/cycle4/mhvk-uncertainty-masked-units-development.md)
Awarded Amount: $54,000.00  (US)
Period of Performance: 2024-05-01 through 2025-04-31 (TBD)
Finance Committee Contact: @aaryapatil
COTR: @mwcraig

**Comment by AnaGabela** [comment]
Hey Marten,
Let's get you started on getting a new contract with NumFocus because I think that's gonna take them a little while. Here are the instructions:

Go to https://numfocus.org/rocket
Click on Independent Contractor Registration
Form fields
Project: Astropy
Grant: NASA
Full Name
Email address
Street Address
Street Address
City
State
Postal Code
Country
Start Date: May 1, 2024-Feb 28, 2025
Description: Specreduce and spectroscopy development
Project Contact name: Astropy Finance Committee
Project Contact email: [finance@astropy.org](mailto:finance@astropy.org)
Payment method: ACH, Wire Transfer, PayPal
Bank Information
Comments about payment
Upload W9 or W8
Please let me know that you have submitted the form so we can keep track of it.

**Comment by mhvk** [comment]
@AnaGabela - thanks, I'm so happy to hear this worked. In the meantime, I've set the wheels in motion to officially reduce my workload at the university of toronto, and will get myself registered soon!

**Comment by mhvk** [comment]
@AnaGabela - sorry for the delay, I've now submitted the relevant information to numfocus. If I understand correctly, just listing that I'm in Canada should ensure no taxes have to be withheld - would be nice to have that confirmed!

Maybe more importantly: I moved the start date to July 1st, to align it with the UofT academic year (and because I will not have much time before that date). I hope that is not a problem.

**Comment by AnaGabela** [comment]
@mhvk- no problem
For the taxes thing all should be ok as long as you send them a W-8Ben (not a W-9)
The start date change is find the only thing is the end date can't change as we need those funds spent to close out the grants. 
Let me know if you have any other question

**Comment by mhvk** [comment]
Thanks! For the time, I think the money is for a total number of hours, right, so it will just mean I'll spend a bit more time per week before April 30, 2025.

**Comment by mhvk** [comment]
@AnaGabela - sorry for another question: my university's admin people have looked into various options, and there are two: I work a reduced load (with reduced pay, benefits, pension) and get compensated for that directly by numfocus, or I get "bought out", i.e., the university of toronto bills numfocus for my time (with zero impact on my pay, benefits, pension). We've started on the former route, but for me the latter would be nicer in that I just don't have to worry about changing how I do my taxes, worry about pensions, etc. Is being billed by the university possible from the numfocus side? (Note: the cost will definitely not be any higher)

**Comment by AnaGabela** [comment]
@mhvk let me ask!

**Comment by mhvk** [comment]
Thanks, that sounds like something that is either going to be very easy (UofT has done it before), or very difficult (they have not, and US tax forms are impossible). I'll ask!

**Comment by AnaGabela** [comment]
For taxes.. it should be form W8Ben and that is simple.. of course being a university maybe not... but that would be the form to start with when they get to that step.

**Comment by mhvk** [comment]
@AnaGabela - UofT seems to expect some form of MoU. That seems reasonable, but I don't know at all if that is possible with NumFocus - is it? (Or if you don't know, do you know who to ask?)

**Comment by mhvk** [comment]
For a MoU, I seem to need the address, e-mail address of the responsible people at NumFocus. Or does this all go through you?

Note that @kelle in earlier discussion wrote that she had had a contract via Barnard (I think); the difference would be that UofT apparently does not charge overhead.

**Comment by kelle** [comment]
In order to start this process, y'all (probably someone at UofT) should complete the ICA registration form.  It asks for the payment details -- how does UofT want to get paid?  

If it turns out an MoU makes more sense instead of a "contract", that's between UofT and the NumFocus folks. Completing the ICA registration from starts this whole process.

**Comment by mhvk** [comment]
@kelle - do you mean the Independent Contractor Registration on https://numfocus.org/rocket? Is there someone specific on the numfocus side that my department administrator can contact? Likely easier without us in the middle!

**Comment by kelle** [comment]
Let me chat with the NumFOCUS folks and get back to you.

**Comment by mhvk** [comment]
@kelle, @AnaGabela  - in the meantime, I got an e-mail from Janay Warner about my initial submission as an independent contractor. Would that be the right person to ask details about how one would go about funding via UofT? 

On the UofT side, the preferred way forward seems to be an MOU, and then UofT invoicing NumFocus three times, at US$18000 each (i.e., no extra costs). Was that kind of process similar to what was done for you, @kelle? Thanks!!!

**Comment by kelle** [comment]
Yes, Janay is the correct person to discuss the MOU with. You may put the UofT people directly in contact with them.

**Comment by mhvk** [comment]
> Yes, Janay is the correct person to discuss the MOU with. You may put the UofT people directly in contact with them.

OK, thanks.  I've sent an inquiry...

**Comment by eteq** [comment]
@mhvk - to jump in here a bit: it looks like it may turn out the direct contractor is the only way to go due to NASA requiring a rebudget for the UofT award approach, which we can't do because we just recently did one.  NF is still working on this to be sure, but I wanted to let you know that *may* be the way we have to go.

In the future, I think we want to make this easier as far as not requiring lots of rebudgets to do this kind of sub-awarding.  But my guess is that's not something we can make happen on the timescale you want to get started.  (It will probably be a *lot* easier if we get the second ROSES award that we've just applied for, but of course we have no idea where that stands and won't for some time yet.)

**Comment by jdswinbank** [comment]
Hey @aaryapatil , @AnaGabela — we've now received an invoice from U. Toronto, but I don't see a (documented) resolution to the discussion above, and I don't think we have an agreement on file (under https://drive.google.com/drive/folders/1VrNzVNGoa0xHGrgk2h5JjD43_pmUiawT). Do you know what the outcome was here? I don't want to approve the invoice until we're sure everybody landed on the same page.

**Comment by AnaGabela** [comment]
oops.. not sure I know.. I was just following the contact part making it through numfocus.. @kelle or @eteq  probably know.

**Comment by jdswinbank** [comment]
Thanks Ana!

If whatever contract NumFOCUS have agreed ends up in the Google Drive linked above, that would probably answer my questions...

**Comment by AnaGabela** [comment]
[contact.pdf](https://github.com/user-attachments/files/17715060/contact.pdf)
This is what I have.

**Comment by kelle** [comment]
Looks like we've got a "services agreement" instead of a subcontract between UT and NumFocus. Great!

**Comment by aaryapatil** [comment]
I signed the service agreement and was waiting for it to go through NumFocus. Looks like everything is set on that end, but I am not sure if anything else is still needed.

**Comment by mhvk** [comment]
Yes, it took a bit of time on both sides, but the service agreement ended up being possible -- which I'm very happy with since that means no hassle with taxes and pensions on my side.

More importantly, the first part -- masked representations and coordinates -- will be in 7.0.  And we just had a nice meeting at flatiron which set a good way forward for Quantity 2.0.

**Comment by jdswinbank** [comment]
Hey @aaryapatil — as the Finance Ctte POC, could you please make sure this issue is up to date with the relevant COTR etc? I think that's @mwcraig but not certain...

**Comment by mwcraig** [comment]
I am the COTR

**Comment by mhvk** [comment]
Following the note from @aaryapatil about extensions to Cycle 4: I think the current project, which was meant to be for three years, is proceeding well, and I hope it can be continued. I last sent an update to @mwcraig on January 24th, the relevant parts of which I paste in below. Let me know if more information is needed! 

> [...] progress since we last wrote has been good.  In particular, the masked representations and skycoords that I described the last time are now fully merged, via https://github.com/astropy/astropy/pull/17016
>
> [...] I am very happy that one of three big chunks of my proposal actually got done, and well ahead of time!  Though that is also because my department gave me teaching release for the Fall semester based on the astropy money coming in, so I could spend well over the average day/week in that period.
>
> After this, I focussed on the Quantity 2.0 project, which got kickstarted with a visit to Adrian and Jeff at Flatiron (with Nathaniel calling in).  Progress there has been good in Nov/Dec, in a new astropy repository, https://github.com/astropy/quantity-2.0/ (And that also got the new SkyCoord draft APE going; https://github.com/astropy/astropy-APEs/pull/112)
> This month, we also got in touch with maintainers of other units packages, with the idea of creating a common API.  This let to the creation of yet another repository, https://github.com/quantity-dev [now getting quite active]
>
> On my side, progress is now slower because this semester I am teaching, and I also am on the graduate admissions committee.  Right now most of my astropy time is spent on more mundane PR review, etc.  This should get a bit better once admissions has finished.
>
> Bottom line: I've already done roughly what I hoped to do in the first year, and believe some further progress is likely.  I very much hope that with the new grant I'll be able to continue similarly for the originally asked full three years.

**Comment by aaryapatil** [comment]
@mhvk , please complete the following contract modification form:

Workday form: https://forms.monday.com/forms/1bb44c839787d25bd6ee4acd456db414?r=use1
Your information:
* **Contractor Name**
* **Contractor Contact**
* **Contact Email address**
* **Full street address**

Contract details:
* **Contract Type:** Amendment
* **Project:** Astropy 

* **Project PI Contact:** @aaryapatil

* **Grant Type:** NASA
* **Grant Reference:** ROSES20 (20-OSTFL20-0057)

* **Hourly Rate:** $18,000 in three installments
* **Contract Value:** $108,000.00 (original + increase)
* **Start Date:** July 1, 2024 (original)
* **End Date:** July 1, 2025 (new)

* **Budget Category:** Development & Maintenance
* **Description of work:** Contractor will contribute to/provide/develop uncertainties, masks, quantities, other types of arrays
* **Statement of Work:** Copy and paste from approved [project description](https://github.com/astropy/astropy-project/blob/main/finance/proposal-calls/cycle4/mhvk-uncertainty-masked-units-development.md).

* **W9 (US workers) or W8 BEN (non-US workers)**

**Comment by mhvk** [comment]
@aaryapatil - are you sure that's correct? At my request, I ended up not being paid directly, but with UofT making a contract with NumFocus (where UofT gets paid for my time, and I just get my regular salary), so I think we'd probably need an extension of that.

**Comment by mhvk** [comment]
I guess more specifically: is `NumFocus` aware of the extension? If so, I can just ask our administrator to contact them and discuss.

**Comment by kelle** [comment]
In this case, the Contractor Name and contact should be the UT admin. - I'll modify the comment above.

The proper workflow for modifying the extension is via submission of this form. This is how one should contact them for this type of request.

**Comment by kelle** [comment]
You could also your admin to submit the form.

**Comment by aaryapatil** [comment]
@mhvk Could you let me know if you have further questions? The process as Kelle described begins with you (or UT admin) requesting an extension of the contract through the form above. After this, NumFocus will be aware of this extension request and will take care of the rest.

**Comment by mhvk** [comment]
@aaryapatil - OK, submitted, after fiddling with the fields to get the form to submit. Note that in lieu of a W8/W9, which is not needed in my case, I uploaded the previous contract.

**Comment by kelle** [comment]
For budgeting reasons, we are only releasing $18,000 in additional funds.  Once these are spent, we can release more, up to the approved budget amount.

**Comment by mhvk** [comment]
@kelle - that sounds fine, but just to be sure: should UofT just continue to invoice every 4 months?

Also, shouldn't there be an official extension of the contract that was signed? (Happy to be told we need to take that up with NumFocus!)

**Comment by kelle** [comment]
Yes, UofT can continue to invoice at their pace. (That is what we are hoping/planning on.)

Yes, it is my understanding that the UT contract amendment is in the queue. From what I can see, it has not been drafted yet.

**Comment by mhvk** [comment]
Super, thanks!!
